### PR TITLE
Update Accessibility Questions

### PR DESCRIPTION
**Pull request by normanbrobinson**
Added accessibility to considerations question. 

Added generic accessibility question with loose coupling to web only example. 

Modified 'ARIA' domain-specific question to clarify expertise in use of Web Accessibility Initiative - Accessible Rich Internet Applications (WAI-ARIA ; specifically for UX designed for assistive technologies instead of anyone in general that may use features build into a user agent) versus baseline accessibility techniques as described in WCAG in general. Previous question equated accessibility to ARIA: the first rule of WAI-ARIA is don't use ARIA unless needed!)

Added system level accessibility framework question to determine if candidate is familiar with how accessibility works at a operating system and window manager level.

Expanded acronym for flash of unstyled content (FOUC).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,9 +44,11 @@ This file contains a number of front-end interview questions that can be used wh
 * If you could master one technology this year, what would it be?
 * What are the differences between Long-Polling, Websockets and SSE?
 * Explain the importance of standards and standards bodies.
-* What is FOUC? How do you avoid FOUC?
+* What is Flash of Unstyled Content (FOUC)? How do you avoid FOUC?
 * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
-* Explain what ARIA and screenreaders are, and how to make a website accessible.
+* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques?
```
Comment (richgilbank) [inline README.md]: This feels a little awkward to me - how about `What is your approach to accessibility with regards to keyboard control and visual focus using standard (non-ARIA) HTML and JavaScript techniques?`

File: README.md [hunk 2]
```diff
@@ -44,9 +44,11 @@ This file contains a number of front-end interview questions that can be used wh
...
 * What are the differences between Long-Polling, Websockets and SSE?
 * Explain the importance of standards and standards bodies.
-* What is FOUC? How do you avoid FOUC?
+* What is Flash of Unstyled Content (FOUC)? How do you avoid FOUC?
 * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
-* Explain what ARIA and screenreaders are, and how to make a website accessible.
+* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques? 
+* Explain your experience with Web Accessibility Initiative's Accessible Rich Internet Applications (WAI-ARIA): Do you know the difference between developing an aceesible website following Web Content Accessibility Guidelines (WCAG) web standards and custom developing for assistive technologies such as screenreaders with WAI-ARIA?
+* What experience do you have in developing in frameworks such as Microsoft Active Accessibility (MSAA), Microsoft UI Automation (UIA), or Java Accessibilty Bridge, or Linux Foundation's IAccessible2?
```
Comment (richgilbank) [inline README.md]: Are these relevant to front-end development?

File: README.md [hunk 3]
```diff
@@ -44,9 +44,11 @@ This file contains a number of front-end interview questions that can be used wh
 * If you could master one technology this year, what would it be?
 * What are the differences between Long-Polling, Websockets and SSE?
 * Explain the importance of standards and standards bodies.
-* What is FOUC? How do you avoid FOUC?
+* What is Flash of Unstyled Content (FOUC)? How do you avoid FOUC?
 * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
-* Explain what ARIA and screenreaders are, and how to make a website accessible.
+* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques? 
+* Explain your experience with Web Accessibility Initiative's Accessible Rich Internet Applications (WAI-ARIA): Do you know the difference between developing an aceesible website following Web Content Accessibility Guidelines (WCAG) web standards and custom developing for assistive technologies such as screenreaders with WAI-ARIA?
```
Comment (richgilbank) [inline README.md]: Mind rewording this a bit? This could probably be shortened and simplified.

**Comment by richgilbank** [comment]
@normanbrobinson Thanks for submitting this, we've been lacking a11y questions IMO. I've added a few notes/questions, mind clarifying those or cleaning them up?

File: README.md (hunk 1 above)
Comment (normanbrobinson) [inline README.md]: I think that is an excellent edit for most interviewers. Thanks for the
review!

On Tue, Jan 20, 2015 at 3:32 PM, Rich Gilbank <notifications@github.com>
wrote:

> In README.md
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226#discussion-diff-23253088>
> :
>
> >  * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
> > -* Explain what ARIA and screenreaders are, and how to make a website accessible.
> > +* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques?
>
> This feels a little awkward to me - how about What is your approach to
> accessibility with regards to keyboard control and visual focus using
> standard (non-ARIA) HTML and JavaScript techniques?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226/files#r23253088>
> .
>



-- 
(703) 942 9569 google voice 24x7
"You have access to the greatest source of knowledge in the universe."
"Well, I do talk to myself sometimes." ~ The Invasion of Time (1978)

File: README.md (hunk 2 above)
Comment (normanbrobinson) [inline README.md]: Absolutely. If they do not have knowledge of these concepts, they have no
experience in addressing accessibility or usability for people with
disabilities. That is to say they probably rely upon "whatever is in the
framework" and don't understand or are honestly ignorant on the topic.

On Tue, Jan 20, 2015 at 3:37 PM, Rich Gilbank <notifications@github.com>
wrote:

> In README.md
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226#discussion-diff-23253470>
> :
>
> >  * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
> > -* Explain what ARIA and screenreaders are, and how to make a website accessible.
> > +* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques?
> > +* Explain your experience with Web Accessibility Initiative's Accessible Rich Internet Applications (WAI-ARIA): Do you know the difference between developing an aceesible website following Web Content Accessibility Guidelines (WCAG) web standards and custom developing for assistive technologies such as screenreaders with WAI-ARIA?
> > +* What experience do you have in developing in frameworks such as Microsoft Active Accessibility (MSAA), Microsoft UI Automation (UIA), or Java Accessibilty Bridge, or Linux Foundation's IAccessible2?
>
> Are these relevant to front-end development?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226/files#r23253470>
> .
>



-- 
(703) 942 9569 google voice 24x7
"You have access to the greatest source of knowledge in the universe."
"Well, I do talk to myself sometimes." ~ The Invasion of Time (1978)

File: README.md (hunk 3 above)
Comment (normanbrobinson) [inline README.md]: Well, I thought of that, but then I thought some of the interviewers might
not have a solid grasp on the topics and thus expanded the formal names
instead of using the informal or acronym only. So I'll vote no for now :)

On Tue, Jan 20, 2015 at 3:41 PM, Rich Gilbank <notifications@github.com>
wrote:

> In README.md
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226#discussion-diff-23253744>
> :
>
> >  * Do your best to describe the process from the time you type in a website's URL to it finishing loading on your screen.
> > -* Explain what ARIA and screenreaders are, and how to make a website accessible.
> > +* What is your approach to accessiblity and developing in keyboard accessibility and visual focus, such as using foundational HTML and Javascript techniques?
> > +* Explain your experience with Web Accessibility Initiative's Accessible Rich Internet Applications (WAI-ARIA): Do you know the difference between developing an aceesible website following Web Content Accessibility Guidelines (WCAG) web standards and custom developing for assistive technologies such as screenreaders with WAI-ARIA?
>
> Mind rewording this a bit? This could probably be shortened and simplified.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226/files#r23253744>
> .
>



-- 
(703) 942 9569 google voice 24x7
"You have access to the greatest source of knowledge in the universe."
"Well, I do talk to myself sometimes." ~ The Invasion of Time (1978)

**Comment by normanbrobinson** [comment]
Happy to assist. Apologies whilst I get into a regular review of the
project and try to make project-appropriate changes; there is so much depth
to this topic that is very hard to over simplify without confusing
usability with accessibility.

On Tue, Jan 20, 2015 at 3:42 PM, Rich Gilbank <notifications@github.com>
wrote:

> @normanbrobinson <https://github.com/normanbrobinson> Thanks for
> submitting this, we've been lacking a11y questions IMO. I've added a few
> notes/questions, mind clarifying those or cleaning them up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226#issuecomment-70728912>
> .
>



-- 
(703) 942 9569 google voice 24x7
"You have access to the greatest source of knowledge in the universe."
"Well, I do talk to myself sometimes." ~ The Invasion of Time (1978)

File: README.md (hunk 3 above)
Comment (darcyclarke) [inline README.md]: What I always try to determine from any new contribution is what kind of skills or knowledge you're trying to vet from a candidate through your question(s). As you note, you want to see if someone has knowledge of accessibility development techniques and standards. That said, the first two questions could, indeed, be simplified to something along the lines of `Do you know how to develop with WAI-ARIA or WCAG standards in mind?` or `What is your experience developing with WAI-ARIA or WCAG standards in mind?`. Acronyms or no acronyms, the rest of the wording is a bit too verbose.

File: README.md (hunk 2 above)
Comment (darcyclarke) [inline README.md]: The last question, as far as I can tell, has nothing to do with front-end development.

Comment (darcyclarke) [inline README.md]: >  If they do not have knowledge of these concepts, they have no
experience in addressing accessibility or usability for people with
disabilities.

I think that's a bit of a broad statement. It's like saying "If you haven't worked with jQuery you've never written JavaScript" (which is untrue of course). Developers can still work to address and enhance the experience of users with disabilities through a number of means and the most prevalent document/guideline we adhere to, from my experience, is the [Web Content Accessibility Guidelines - WCAG](http://www.w3.org/WAI/intro/wcag). 

I do have to admit and plead ignorance in dealing with these specific technologies myself. That said, I did some quick research and it looks like many of these frameworks deal with the browser vendor implementations and not specifically author's (ie. front-end developer's) experiences. I could be wrong but I couldn't find any recommendations or guidelines for how I, as a front-end developer, would better adhere or integrate with things like `Microsoft UI Automation`. I did, however, find some helpful MDN documents which note Firefox's [integration](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Implementing_MSAA_server) and [adherence](https://developer.mozilla.org/en-US/docs/Web/Accessibility/AT-APIs) with MSAA etc.. Java applications, notwithstanding, I agree with @richgilbank that these don't seem relevant. Feel free to correct me if I'm wrong.

File: README.md (hunk 3 above)
Comment (normanbrobinson) [inline README.md]: I suppose there needs to be a better way to explain the purpose of why the question is unique and the goal of the question. There are many people who won't know WAI-ARIA, nor WCAG. The first question is generic: what does one know about accessibility? The second question is specific to (1) do you have experience with WAI-ARIA and (2) based on getting feedback on what they might know about accessibility in general, can they explain the differences between accessibility (foundational HTML techniques via WCAG, Section 508, etc.) and WAI-ARIA which is very specific to supporting assistive technologies such as screen reader usage (that is not so much about accessibility, as much as accommodation, user experience and use cases for user interface design patterns that need to be linearized or modalized for use with a specific product).

File: README.md (hunk 2 above)
Comment (normanbrobinson) [inline README.md]: The question was, are the relevant to front-end development. My answer, absent your suggesting the response to richgilbank was overly broad, is correct. 

You yourself just brought it around to the expected response to the question "What is your approach to accessibility...": adhering to WCAG would be a good thing to here as a leading response.

Specifically for web browsers, the next level of user experience is having knowledge of the assistive technologies (screen readers, magnifiers, etc.) use and how they are used to understand what design patterns or approaches are needed to support them, and what additional tailoring is possible (WAI-ARIA) to include why that can actually be detrimental to your end users (WAI-ARIA override of name, role, state of foundational HTML elements).

Framework is a generic term. Any framework will do. Abstract away! My point is people without the experience assume the framework has accessibility built in or don't know about accessibility so never asked the question, and don't have the experience. They might be smart enough to figure it out, but these are _interview_ questions.

Most of these simple questions have been targeted at front-end development of web browser applications. Although the Access Board's (Section 508) standards for accessibility are drafted and equate many of these accessibility functions currently implemented in web applications in software concepts.

The highest level of understanding of how accessibiltiy affects the end user is understanding the operating systems build in framework (e.g., Microsoft historically using MSAA and now UIA) and, by example, how browser manufacturers use that framework to deliver accessibility to the operating system, or how assistive technologies such as screen readers rely on those same APIs and frameworks to function.

I thought it too advanced to put in this sort of initial questioning, but an developer experienced in accessibility with this background would be able to discern why Firefox's current approach to the operating system accessibility on Windows 8 might fail to work for a screen reader user, even if coded properly. They should also understand how to use the UIA framework to code their own (C, C++, .NET, any language will do) application to insure it is accessible to the operating system and assistive technologies for that platform. By extension they should also understand the general approach to accessibility used, and determine the platform specific accessibility layers indicates iOS is different from Andriod, is different from Linux+Window manager, is different from Windows (MSAA for earlier versions, UIA since Windows 7).

I hope that helps explain my perspective. Written responses isn't always the easiest medium to explain nor provide emotional intent to the response, so please forgive me if my attempt to clarify seems curt. I only wanted to respect your time and try and be clear. I sincerely thank you for taking the time to be critical of my suggestions, as you've also given me a chance to consider how to better introduce the topic to people like yourself that may have the same concerns. Please continue to help me make this better!

Comment (darcyclarke) [inline README.md]: Thanks for the in depth response @normanbrobinson. That said, I still don't think this is a typical front-end developer interview question. 

As a way to move forward, I believe my suggestion for [line 50](https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226/files#r23285098) nicely homogenizes your accessibility concerns.

**Comment by darcyclarke** [comment]
@normanbrobinson Based on the discussions in-line, I think there's a few revisions that need to be made before we can merge this. It would also be great if you could re-title your pull request to reflect the changes your wishing to make (see [Contributing](https://github.com/h5bp/Front-end-Developer-Interview-Questions/blob/master/CONTRIBUTING.md) for more details).

**Comment by darcyclarke** [comment]
@normanbrobinson This branch is now too far out of date to be merged cleanly. If you could rebase or merge and clean up/fix up this PR that would be great. I'm going to close this PR for now though but can reopen once those changes are made or you can make a new PR if you'd like.

**Comment by normanbrobinson** [comment]
Sorry - had two deaths in the family - will (re)do asap

On Tue, Feb 17, 2015 at 6:20 PM, Darcy Clarke <notifications@github.com>
wrote:

> @normanbrobinson <https://github.com/normanbrobinson> This branch is now
> too far out of date to be merged cleanly. If you could rebase or merge and
> clean up/fix up this PR that would be great. I'm going to close this PR for
> now though but can reopen once those changes are made or you can make a new
> PR if you'd like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5bp/Front-end-Developer-Interview-Questions/pull/226#issuecomment-74776993>
> .
>



-- 
(703) 942 9569 google voice 24x7
"You have access to the greatest source of knowledge in the universe."
"Well, I do talk to myself sometimes." ~ The Invasion of Time (1978)
